core/drivers/pl022_spi.c

```diff
@@ -273,10 +265,11 @@ static void pl022_sanity_check(struct pl022_data *pd)
 {
 	assert(pd);
 	assert(pd->chip.ops);
-	assert(pd->gpio);
-	assert(pd->gpio->ops);
-	assert(pd->base);
-	assert(pd->cs_gpio_base);
+	/* pd->cs_data.cs_cb CAN be NULL so no assert */
+	if (!pd->b_cs_callback) {
```

Comment:
Because of the union in the structure, can't distinguish between 
Maybe the union is useless. Or maybe a "default" callback for acting on a gpio could be used so that is follows the same scheme! already call the CS callback handler.